Import menu entry does not work

The calendar (+) button -> **Import** does not work.

Error in the console:
```
angular-strap.js:2501 Error: [$injector:unpr] Unknown provider: davImportServiceProvider <- davImportService <- CalCalendarImportController
http://errors.angularjs.org/1.3.20/$injector/unpr?p0=davImportServiceProvider%20%3C-%20davImportService%20%3C-%20CalCalendarImportController
```

--- a/package.json
+++ b/package.json
@@ -14,6 +14,7 @@
   "author": "",
   "license": "AGPLv3",
   "dependencies": {
+    "esn-dav-import-client": "github:openpaas-suite/esn-dav-import-client#main",
     "esn-frontend-common-libs": "github:openpaas-suite/esn-frontend-common-libs#main",
     "esn-frontend-mailto-handler": "github:openpaas-suite/esn-frontend-mailto-handler#main",
     "esn-frontend-videoconference-calendar": "github:openpaas-suite/esn-frontend-videoconference-calendar#main",

--- a/src/linagora.esn.calendar/app/app.js
+++ b/src/linagora.esn.calendar/app/app.js
@@ -12,6 +12,7 @@ angular.module('esn.calendar', [
   'esn.calMoment',
   'esn.configuration',
   'esn.core',
+  'esn.file',
   'esn.form.helper',
   'esn.header',
   'esn.highlight',
@@ -67,6 +68,7 @@ require('esn-frontend-common-libs/src/frontend/js/modules/i18n/i18n.module.js');
 require('esn-frontend-common-libs/src/frontend/js/modules/session.js');
 require('esn-frontend-common-libs/src/frontend/js/modules/scroll.js');
 require('esn-frontend-common-libs/src/frontend/js/modules/core.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/file.js');
 require('esn-frontend-common-libs/src/frontend/js/modules/datetime/datetime.module.js');
 require('esn-frontend-common-libs/src/frontend/components/openpaas-logo/openpaas-logo.js');
 require('esn-frontend-common-libs/src/modules/linagora.esn.graceperiod/frontend/js/app.js');

--- a/src/linagora.esn.calendar/app/settings/import/calendar-import.controller.js
+++ b/src/linagora.esn.calendar/app/settings/import/calendar-import.controller.js
@@ -1,12 +1,14 @@
 'use strict';
 
+const { ESNDavImportClient } = require('esn-dav-import-client');
+
 angular.module('esn.calendar')
   .controller('CalCalendarImportController', CalCalendarImportController);
 
 function CalCalendarImportController(
   $window,
   asyncAction,
-  davImportService,
+  fileUploadService,
   calendarService,
   calendarHomeService,
   calUIAuthorizationService,
@@ -48,7 +50,10 @@ function CalCalendarImportController(
   }
 
   function importFromFile() {
-    return davImportService.importFromFile(self.file, self.calendar.href);
+    const OPENPAAS_URL = process.env.OPENPAAS_URL || 'http://localhost:8080';
+    const esnDavImportClient = new ESNDavImportClient(fileUploadService.uploadFile, OPENPAAS_URL);
+
+    return esnDavImportClient.importFromFile(self.file, self.calendar.href);
   }
 
   function submit() {

--- a/src/linagora.esn.calendar/app/settings/import/calendar-import.controller.js
+++ b/src/linagora.esn.calendar/app/settings/import/calendar-import.controller.js
@@ -1,76 +1,74 @@
-(function(angular) {
-  'use strict';
+'use strict';
 
-  angular.module('esn.calendar')
-    .controller('CalCalendarImportController', CalCalendarImportController);
+angular.module('esn.calendar')
+  .controller('CalCalendarImportController', CalCalendarImportController);
 
-  function CalCalendarImportController(
-    $window,
-    asyncAction,
-    davImportService,
-    calendarService,
-    calendarHomeService,
-    calUIAuthorizationService,
-    session
-  ) {
-    var self = this;
+function CalCalendarImportController(
+  $window,
+  asyncAction,
+  davImportService,
+  calendarService,
+  calendarHomeService,
+  calUIAuthorizationService,
+  session
+) {
+  var self = this;
 
-    self.$onInit = $onInit;
-    self.canModifyCalendar = canModifyCalendar;
-    self.onFileSelect = onFileSelect;
-    self.file = null;
-    self.isValid = null;
-    self.submit = submit;
+  self.$onInit = $onInit;
+  self.canModifyCalendar = canModifyCalendar;
+  self.onFileSelect = onFileSelect;
+  self.file = null;
+  self.isValid = null;
+  self.submit = submit;
 
-    $onInit();
+  $onInit();
 
-    ///////////
+  ///////////
 
-    function $onInit() {
-      calendarHomeService.getUserCalendarHomeId()
-        .then(calendarService.listPersonalAndAcceptedDelegationCalendars)
-        .then(function(calendars) {
-          self.calendar = calendars[0];
-          self.calendars = calendars;
-        });
-    }
-
-    function onFileSelect(file) {
-      if (!file || !(file.length > 0)) {
-        return;
-      }
-
-      self.file = file[0];
-      self.isValid = self.file.type === 'text/calendar';
-    }
-
-    function canModifyCalendar(calendar) {
-      return calUIAuthorizationService.canImportCalendarIcs(calendar, session.user._id);
-    }
-
-    function importFromFile() {
-      return davImportService.importFromFile(self.file, self.calendar.href);
-    }
-
-    function submit() {
-      var notificationMessages = {
-        progressing: 'Submitting importing calendar request...',
-        success: 'Request submitted',
-        failure: 'Failed to submit request'
-      };
-
-      var reloadOption = {
-        onSuccess: {
-          linkText: 'Reload',
-          action: function() { $window.location.reload(); }
-        }
-      };
-
-      asyncAction(
-        notificationMessages,
-        importFromFile,
-        reloadOption
-      );
-    }
+  function $onInit() {
+    calendarHomeService.getUserCalendarHomeId()
+      .then(calendarService.listPersonalAndAcceptedDelegationCalendars)
+      .then(function(calendars) {
+        self.calendar = calendars[0];
+        self.calendars = calendars;
+      });
   }
-})(angular);
+
+  function onFileSelect(file) {
+    if (!file || !(file.length > 0)) {
+      return;
+    }
+
+    self.file = file[0];
+    self.isValid = self.file.type === 'text/calendar';
+  }
+
+  function canModifyCalendar(calendar) {
+    return calUIAuthorizationService.canImportCalendarIcs(calendar, session.user._id);
+  }
+
+  function importFromFile() {
+    return davImportService.importFromFile(self.file, self.calendar.href);
+  }
+
+  function submit() {
+    var notificationMessages = {
+      progressing: 'Submitting importing calendar request...',
+      success: 'Request submitted',
+      failure: 'Failed to submit request'
+    };
+
+    var reloadOption = {
+      onSuccess: {
+        linkText: 'Reload',
+        action: function() { $window.location.reload(); }
+      }
+    };
+
+    asyncAction(
+      notificationMessages,
+      importFromFile,
+      reloadOption
+    );
+  }
+}

--- a/src/linagora.esn.calendar/app/settings/import/calendar-import.controller.spec.js
+++ b/src/linagora.esn.calendar/app/settings/import/calendar-import.controller.spec.js
@@ -3,7 +3,8 @@
 /* global chai: false */
 /* global sinon: false */
 
-var expect = chai.expect;
+const { expect } = chai;
+const { ESNDavImportClient } = require('esn-dav-import-client');
 
 describe('The CalCalendarImportController', function() {
 
@@ -182,7 +183,17 @@ describe('The CalCalendarImportController', function() {
   });
 
   describe('The submit function', function() {
-    it('should call davImportService.importFromFile with selected file and selected calendar', function() {
+    let sandbox;
+
+    beforeEach(function() {
+      sandbox = sinon.sandbox.create();
+    });
+
+    afterEach(function() {
+      sandbox.restore();
+    });
+
+    it('should call ESNDavImportClient.importFromFile with selected file and selected calendar', function() {
       session.user._id = '123';
 
       var calendars = [{
@@ -201,6 +212,8 @@ describe('The CalCalendarImportController', function() {
         return $q.when(session.user._id);
       });
 
+      sandbox.stub(ESNDavImportClient.prototype, 'importFromFile').returns($q.when());
+
       calendarService.listPersonalAndAcceptedDelegationCalendars = sinon.stub().returns($q.when(calendars));
 
       var controller = initController();
@@ -209,13 +222,11 @@ describe('The CalCalendarImportController', function() {
         length: 100
       }];
 
-      davImportService.importFromFile = sinon.stub().returns($q.when());
-
       controller.onFileSelect(file);
       controller.submit();
       $rootScope.$digest();
 
-      expect(davImportService.importFromFile).to.have.been.calledWith(controller.file, calendars[0].href);
+      expect(ESNDavImportClient.prototype.importFromFile).to.have.been.calledWith(controller.file, calendars[0].href);
     });
   });
 });

